### DISCUSSION TITLE
ipv6 leak

### DISCUSSION THREAD

**Discussion by Alistair1231** [wiresock/WireSockUI#38] [Ideas]
Hi. Great app, however my ipv6 leaks. I tested by adding `msedge` to the allowed apps and opening https://ipleak.net. DNS and IPv4 are fine but IPv6 isn't. The vpn provider doesn't support ipv6 and somehow it falls back to my main adapter. I even tried disabling IPv6 in the Wiresock Network adapter but that doesn't change anything. What does work is disabling IPv6 on my main network adapter, but I would prefer to not do that.
![image](https://github.com/wiresock/WireSockUI/assets/26165288/facb5635-a7ec-49a1-89d7-af295f9c68c5)
I did try to add `::/0` to the AllowedIPs but it doesn't make a difference

![image](https://github.com/wiresock/WireSockUI/assets/26165288/6167aa73-cb3b-4f91-bd9f-1b98d746e12c)

**Comment by wiresock** [comment]
Hi,

Thanks for raising the question about IPv6 support in WireSock VPN Client. Could you please tell us what you expect to happen when your ISP supports IPv6 but your VPN provider doesn't? In cases where there is no IPv6 address in the configuration, the IPv6 traffic will just go over the default ISP connection.

Regarding a workaround, if you would like your IPv6 traffic blocked, you might consider adding a fake IPv6 address in your settings. This method will allow IPv6 traffic to be sent over the tunnel, but it will be dropped by the VPN provider. This can be a useful approach if you want to prevent IPv6 traffic from bypassing the VPN tunnel and going directly through your ISP.

We look forward to hearing your thoughts on this and any further details you can provide about your expected behavior in this scenario.

Thanks,
Vadim

**Comment by Alistair1231** [comment]
Great idea. Thanks, this seems to work! For anyone wondering what I changed (the ipv6 address can be anything):
![image](https://github.com/wiresock/WireSockUI/assets/26165288/22735d98-b0d7-49bc-bbc5-19a512784494)
